### PR TITLE
CLI: Restrict file permissions on Windows

### LINKED ISSUE #624: lower permissions file with mnemonic on windows
## Description
Windows has a much different permission model than POSIX, (UNIX).
Files containing the mnemonic should have been saved by CLI with permissions similar to 600 on the UNIX system. 

https://github.com/iotaledger/iota-sdk/pull/611#discussion_r1231960169

## Motivation

Its a good security practice to use a layered security model

## Requirements

Write a list of what you want this feature to do.

1.  the ile containing a mnemonic shouldn't be read by other users or groups of users than the user that starts the process 

## Open questions (optional)

what is the secure equivalence of 0600 on Windows?

## Are you planning to do it yourself in a pull request?

Yes/No.

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

CLI: Restrict file permissions on Windows

## Links to any relevant issues

Closes #624 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Running the cli wallet
The rights look now like this
image
Without it would look like this
image

### CODE REVIEW

File: cli/Cargo.toml [hunk 1]
```diff
@@ -44,3 +44,7 @@ serde_json = { version = "1.0.107", default-features = false }
 thiserror = { version = "1.0.48", default-features = false }
 tokio = { version = "1.32.0", default-features = false, features = ["fs"] }
 zeroize = { version = "1.6.0", default-features = false }
+
+[target.'cfg(target_os = "windows")'.dependencies]
+windows-acl = { version = "0.3.0", default-features = false }
+winapi = { version = "0.3.9", default-features = false }
```
Comment (Thoralf-M) [inline cli/Cargo.toml]:
```suggestion
winapi = { version = "0.3.9", default-features = false }
windows-acl = { version = "0.3.0", default-features = false }
```

File: cli/src/helper.rs [hunk 2]
```diff
@@ -211,6 +211,88 @@ async fn write_mnemonic_to_file(path: &str, mnemonic: &str) -> Result<(), Error>
...
+}
+
+// Slightly modified from https://github.com/sile/sloggers/blob/master/src/permissions.rs
+#[cfg(windows)]
+pub fn restrict_file_permissions<P: AsRef<Path>>(path: P) -> std::io::Result<()> {
+    use winapi::um::winnt::{FILE_GENERIC_READ, FILE_GENERIC_WRITE, STANDARD_RIGHTS_ALL};
+
+    /// This is the security identifier in Windows for the owner of a file. See:
+    /// - https://docs.microsoft.com/en-us/troubleshoot/windows-server/identity/security-identifiers-in-windows#well-known-sids-all-versions-of-windows
+    #[cfg(windows)]
```
Comment (thibault-martinez) [inline cli/src/helper.rs]: We're already under a windows flag, is this necessary?

File: cli/src/helper.rs [hunk 3]
```diff
@@ -211,6 +211,88 @@ async fn write_mnemonic_to_file(path: &str, mnemonic: &str) -> Result<(), Error>
...
+    const OWNER_ACL_ENTRY_MASK: u32 = FILE_GENERIC_READ | FILE_GENERIC_WRITE | STANDARD_RIGHTS_ALL;
+
+    use std::io;
+
+    use winapi::um::winnt::PSID;
+    use windows_acl::{
+        acl::{AceType, ACL},
+        helper::sid_to_string,
+    };
```
Comment (thibault-martinez) [inline cli/src/helper.rs]: Put all imports together at the top of the function

File: cli/src/helper.rs (hunk 2 above)
Comment (Thoralf-M) [reply cli/src/helper.rs]: Thanks, wanted to do that, but forgot

**Review by kwek20 — DISMISSED** [review]

**Review by thibault-martinez — COMMENTED** [review]
Changelog worthy ?

**Review by thibault-martinez — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**